### PR TITLE
Toggle view option for image added in the sidebar

### LINKED ISSUE #724: Add toggle view for images in the sidebar
**Description**

Add toggle view option in the image sidebar to view images in form of a 
- list
- grid 

According to user for a better UI.
Selection_187

### PR DESCRIPTION
**Pull request by divyabaid16**
Fixes #724  (<=== Add issue number here)

Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

GIF:
Peek 2019-06-24 18-15

[//]: # (To mark checkbox write 'x' within the square brackets)

* [x] PR is descriptively titled 📑 and links the original issue above 🔗
* [ ] tests pass -- look for a green checkbox ✔️ a few minutes after opening your PR -- or run tests locally with `rake test`
* [x] code is in uniquely-named feature branch and has no merge conflicts 📁
* [x] screenshots/GIFs are attached 📎 in case of UI updation
* [ ] ask `@publiclab/mapknitter-reviewers` for help, in a comment below

> We're happy to help you get this ready -- don't be afraid to ask for help, and **don't be discouraged** if your tests fail at first!

If tests do fail, click on the red `X` to learn why by reading the logs.

Please be sure you've reviewed our contribution guidelines at https://publiclab.org/contributing-to-public-lab-software 

Thanks!

### CODE REVIEW

**Comment by divyabaid16** [comment]
@publiclab/mapknitter-reviewers

**Comment by divyabaid16** [comment]
All the tests have passed here.
Thanks!

**Comment by kaustubh-nair** [comment]
Looks good to me :+1:

**Comment by gauravano** [comment]
Hi @divyabaid16, looks great. Please remove the merge conflicts. 

As a follow-up, we can improve the design from button to icons? Also, how about showing image options in grid view too? @publiclab/mapknitter-reviewers do you think it would look good?

Thank you.

**Comment by divyabaid16** [comment]
@gauravano I have rebased my PR.
Thanks!

File: app/views/images/_index.html.erb [hunk 1]
```diff
@@ -9,9 +13,10 @@
   </div>
 </div>
 </div>
+
 <% if @map.warpables.empty? %>
   <p id="no-images">No images uploaded. </p>
-  <%end%>
+<%end%>
```
Comment (gauravano) [inline app/views/images/_index.html.erb]:
```suggestion
<% end %>
```

File: app/assets/stylesheets/sidebar.css [hunk 2]
```diff
@@ -1,7 +1,15 @@
...
+  margin-bottom: 10px;
+  display: inline-block;
+  padding-right: 30px;
+  color: orange;
+}
 .image-order{
 	display: inline-block;
 	float: right;
-	margin-right: 35px;
+  float: right;
```
Comment (gauravano) [inline app/assets/stylesheets/sidebar.css]:
```suggestion
        float: right;
```

**Comment by gauravano** [comment]
Hey @divyabaid16, CSS is not working well for images of different size

Screenshot from 2019-07-28 22-54-45

Could you try to limit all the columns of sidebar in a single row? Thank you!

**Comment by divyabaid16** [comment]
List view is now working fine @gauravano .

**Comment by divyabaid16** [comment]
I'm not sure why the test is failing here. Can anyone tell me the reason for this? @publiclab/mapknitter-reviewers

**Comment by cesswairimu** [comment]
Hey @divyabaid16, CodeCov is misbehaving a lot on PRs...Could you try rebasing with main? this might fix it.

**Comment by divyabaid16** [comment]
Hmm, it failed even after rebasing.

**Comment by cesswairimu** [comment]
CodeCov expects us to add tests though you did not change any functionality :thinking:

**Comment by divyabaid16** [comment]
Should we merge this even though the test is failing?

**Comment by jywarren** [comment]
Let's ignore CodeCov here for now, we're working on that elsewhere... can you resolve conflicts and then we should be good? Thanks!

**Comment by divyabaid16** [comment]
@jywarren I have resolved the conflicts. This is good to go.
Thanks!

**Comment by jywarren** [comment]
Nice!!! Great work!

**Pull request merged**